R/functions-xcmsSwath.R

```diff
@@ -0,0 +1,33 @@
+## function "align" EICs
+.alignEics <- function(x, y) {
+
+  # linear interpolation of y on x
+  y_aligned <- approx(rtime(y), intensity(y), rtime(x))
+  
+  # deal with NA
+  y_aligned[is.na()] <- 0
+  
+  # correct rtime and int in chromatogram object
+  y@rtime <- y_aligned$x
+  y@intensity <- y_aligned$y
+  
+  # return
+  return(y)
+  
+}
+
+# function to correlate EICs
+.correlateEics <- function(x, y) {
+  
+  # check if x and y are on the same time scale
+  if(!rtime(x) == rtime(y)) {
+    y <- .alignEics(x, y)
+  }
+  
+  # correlate values
+  corValues <- cor.test(intensity(x), intensity(y))
```

Comment:
Do we use also the p-values? I think we can live with *just* the correlation coefficient and then we could use `cor` instead and return a single value instead of a `list` (would make later processing easier of the result also).

Reply:
I used them sometimes. Agree, single value is easier to handle